### PR TITLE
Adding support for dictionaries, in addition to arrays

### PR DESCRIPTION
**Pull request by mabedan**

### CODE REVIEW

File: JSONHelper/JSONHelper.swift [hunk 1]
```diff
@@ -316,6 +336,167 @@ public func <<<* (inout array: [NSDate], value: AnyObject?) -> [NSDate] {
...
+  } else {
+    dictionary = nil
+  }
+  return dictionary
+}
+
+public func <<<* (inout dictionary: [String: Double], value: AnyObject?) -> [String: Double] {
+  var newValue: [String: Double]?
+  newValue <<<* value
+  if let newValue = newValue { dictionary = newValue }
```
Comment (amasry) [inline JSONHelper/JSONHelper.swift]: what does this line mean? :D

**Comment by isair** [comment]
Thanks for adding support for dictionaries. So it seems adding support for dictionaries can't be shortened just like arrays. I need to look into finding a way to do that soon, if it's possible.

**Comment by isair** [comment]
Looks like I've been a bit hasty at merging it. Morning mistake. It seems the swift compiler gets confused with <<<* after adding support for dictionaries. Tests fail to even compile on Xcode 6.1.1 (6A2006). Are you facing this issue as well?

**Comment by mabedan** [comment]
My tests are green... maybe you need to empty your [derived data](http://stackoverflow.com/questions/18933321/deleting-contents-from-xcode-derived-data-folder). In any case, there's a spelling mistake in readme.md

**Comment by isair** [comment]
I can confirm that the problem persists on my machine even after cleaning my derived data. The exact errors can be seen in this screenshot: https://www.dropbox.com/s/a30vw858xwx7y2h/Screenshot%202015-01-30%2013.13.13.png?dl=0

**Comment by mabedan** [comment]
Interesting... my version is 6.1.1 (6A2008a)

**Comment by isair** [comment]
I downloaded 6.1.1 from the app store, which is build 6A2008a, over my 6.1.1 GM release. I was pretty sure that would fix it, however it did not. I'll try a couple more things and let you know.

**Comment by mabedan** [comment]
Maybe an OS issue? 
Mine is 10.10.2 (14C109)

**Pull request merged**